charts/kubewarden-controller/templates/cert-tls.yaml

```diff
@@ -0,0 +1,29 @@
+# cert-manager resources
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ include "kubewarden-controller.fullname" . }}-serving-cert
+  namespace: {{ .Release.Namespace }}
+spec:
+  dnsNames:
+  - {{ include "kubewarden-controller.fullname" . }}-webhook-service.{{ .Release.Namespace }}.svc
+  - {{ include "kubewarden-controller.fullname" . }}-webhook-service.{{ .Release.Namespace }}.svc.cluster.local
+  issuerRef:
+    kind: Issuer
+{{- if eq .Values.tls.source "cert-manager-self-signed" }}
+    name: {{ include "kubewarden-controller.fullname" . }}-selfsigned-issuer
+{{- else if eq .Values.tls.source "cert-manager" }}
+    name: {{ .Values.tls.certManagerIssuerName }}
+{{- end}}
```

Comment:
I think this is a bit "complicated", it can also lead to unexpected results.

Can't we simply replace all of that with (I suck with go templating, but I guess you get the idea of what I mean):

```
{{- if eq (len .Values.tls.certManagerIssuerName) 0 }}
    name: {{ include "kubewarden-controller.fullname" . }}-selfsigned-issuer
{{- else  }}
    name: {{ .Values.tls.certManagerIssuerName }}
{{- end}}
```

and then we simply drop the `tls.source` variable

Reply:
I see the point; I'm just aiming for a values schema that is more future proof.

Thinking about future changes, I suppose the next one could be adding an option use `tls.cert` and `tls.key`:
Without `tls.source == from-var` both UX and templating would get even more complicated. 
Setting `tls.CertManagerIssuerName`, `tls.cert`, `tls.key` at the same time would be ambiguous, and the default (cert-manager self-signed) is obscured. Also, I think is better for semver if we cleanly allow for the values schema to be expanded, without needing to bump a major version (simplifying either `helm upgrade` or our own logic to set defaults).

I would be glad to have a tiebreaker here :).

Reply:
As discussed during the daily, I'm fine with your approach. Let's go ahead with the current PR